### ISSUE TITLE
bootstrap VS foundation VS vanilla CSS ??

### ISSUE THREAD

**Issue by relvao** [getlackey/cms#50]
We have been shifting between bootstrap and foundation. What do you want to use as default?

As it stands this site has html for both of those as we did some copy/paste. That will need fixing.

**Comment by robmccardle** [comment]
Bootstrap. Please remove all the Foundation code.

**Comment by SteffanDonal** [comment]
I disagree, and believe that we should use Foundation through SASS as a framework.

Twitter Bootstrap has many advantages, mainly being fast to set up, responsive, and easy to customise. It's designed for getting a quick app off of the ground with minimal effort on CSS boilerplate, not as a base for a long-lived, mature application.

Foundation is a good compromise away from Bootstrap; it doesn't make assumptions about how the page should look or be structured, and instead has you add classes to the markup to represent layout. It's nice as it allows for easy targeted responsive styling whilst being flexible for your own styling and theming.

However both Twitter Bootstrap and Foundation have one major downside that is going to become more and more important in the near future.
They pollute the markup with style-only classes and structure. Semantic HTML is an important concept that is currently largely ignored, unless it's about tables. A long time ago we stopped using tables for layout, because tables are for tabular data, not layout. The same is now being pushed for the rest of the page.

The idea is to keep the HTML markup as style-less as possible, to allow for greater flexibility in the future, better readability, and improved accessibility. For example, a common Foundation class is: "col-sm-pull-4 col-sm-8 col-md-pull-3 col-md-6". This tells the person maintaining your code exactly nothing about the content inside - and requires that they have existing knowledge of how to structure some element or portion of the site.

The proposal is to use Foundation's SASS support to clean up the markup, to make it framework-independent, and then move all styling into CSS.

This will give us greater flexibility with layout, responsiveness, and accessibility.

**Comment by robmccardle** [comment]
All valid/interesting points; the pro Bootstrap arguments mainly came from Tim D so I'm flexible here and if we can remain framework agnostic that's even better. Let's have a chat around this. Cheers
